src/components/settings/SettingsLMSTab/ErrorReporting/ContentMetadataTable.jsx

```diff
@@ -9,38 +11,81 @@ import DownloadCsvButton from './DownloadCsvButton';
 import { getSyncStatus, getSyncTime } from './utils';
 
 function ContentMetadataTable({ config, enterpriseCustomerUuid }) {
-  const [data, setData] = useState([]);
+  const [currentPage, setCurrentPage] = useState();
+  const [currentFilters, setCurrentFilters] = useState();
+  const [paginationData, setPaginationData] = useState({
+    itemCount: 0,
+    pageCount: 0,
+    data: [],
+    sortBy: '',
+  });
 
   useEffect(() => {
     const fetchData = async () => {
       const response = await LmsApiService.fetchContentMetadataItemTransmission(
-        enterpriseCustomerUuid, config.channelCode, config.id,
+        enterpriseCustomerUuid, config.channelCode, config.id, currentPage, currentFilters,
       );
       return response;
     };
 
     fetchData()
       .then((response) => {
-        setData(response.data.results);
+        setPaginationData({
+          itemCount: response.data.count,
+          pageCount: response.data.pages_count,
+          data: response.data.results,
+        });
       })
       .catch((err) => {
         logError(err);
       });
-  }, [config.channelCode, config.id, enterpriseCustomerUuid]);
+  }, [config.channelCode, config.id, enterpriseCustomerUuid, currentPage, currentFilters]);
+
+  // Call back function, handles filters and page changes
+  const fetchData = useCallback(
+    (args) => {
+      const newFiltersMap = args.filters.map((filter) => ({ [filter.id]: filter.value }));
+      const newFilters = Object.assign({}, ...newFiltersMap);
+
+      const newSortByMap = args.sortBy.map((filter) => ({ sort_by: `${filter.desc ? '-' : ''}${filter.id}` }));
+      const newSortBys = Object.assign(newFilters, ...newSortByMap);
+
+      setCurrentFilters({ ...newFilters, ...newSortBys });
+
+      // pages index from 1 in backend, DataTable component index from 0
+      if (args.pageIndex !== currentPage - 1) {
+        setCurrentPage(args.pageIndex + 1);
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
unfortunately yes, because this method updates `currentPage` conditionally based on the value of `currentPage`, if we add it to the dependencies it will continuously run and crash the page. It's a symptom of the very strange behavior I experienced with our usage of Paragon's DataTable.

Reply:
nevermind! I was totally wrong and it's not needed